src/openssl/ssl/hostname_validation.cr

```diff
@@ -0,0 +1,128 @@
+require "socket"
+require "openssl"
+
+# :nodoc:
+module OpenSSL::SSL::HostnameValidation
+  enum Result
+    Error
+    MalformedCertificate
+    MatchFound
+    MatchNotFound
+    NoSANPresent
+  end
+
+  # Matches hostname against Subject Alternate Name (SAN) or Common Name (CN)
+  # entries of certificate if no SAN entries could be found in the certificate.
+  def self.validate_hostname(hostname : String, server_cert : LibCrypto::X509)
+    return Result::Error if server_cert.null?
+    result = matches_subject_alternative_name(hostname, server_cert)
+    result = matches_common_name(hostname, server_cert) if result.no_san_present?
```

Comment:
Looks like that's a fairly recent change http://stackoverflow.com/a/5937270/2199687

But okay with me then.